src/main/java/de/tum/in/www1/artemis/repository/TextBlockRepository.java

```diff
@@ -24,5 +25,17 @@
     @EntityGraph(type = LOAD, attributePaths = { "cluster" })
     List<TextBlock> findAllWithEagerClusterBySubmissionId(Long id);
 
+    /*
+    @Query("SELECT distinct b FROM TextBlock b " +
+        "LEFT JOIN FETCH Submission s on b.submission = s " +
+        "LEFT JOIN Participation p ON s.participation = p " +
+        "LEFT JOIN TextExercise e ON p.exercise = e " +
+        "WHERE e.id = :#{#exerciseId} AND b.treeId IS NOT NULL")
+    List<TextBlock> findAllByTreeIdExistsAndExerciseId(@Param("id") Long exerciseId);
+     */
+
+    // TODO: Test this query
+    List<TextBlock> findAllBySubmission_Participation_Exercise_IdAndTreeIdNotNull(Long exerciseId);
```

Comment:
Codacy found an issue: [The instance method name 'findAllBySubmission_Participation_Exercise_IdAndTreeIdNotNull' doesn't match '[a-z][a-zA-Z0-9]*'](https://app.codacy.com/gh/ls1intum/Artemis/pullRequest?prid=5968857)